modules/imgcodecs/test/test_exr.impl.hpp

```diff
@@ -113,5 +125,35 @@ TEST(Imgcodecs_EXR, readWrite_32FC3_half)
     EXPECT_EQ(0, remove(filenameOutput.c_str()));
 }
 
+TEST(Imgcodecs_EXR, readWrite_32FC1_PIZ)
+{
+  const string root = cvtest::TS::ptr()->get_data_path();
+  const string filenameInput = root + "readwrite/test32FC1.exr";
+  const string filenameOutput = cv::tempfile(".exr");
+
+  std::vector<int> params;
+  params.push_back(IMWRITE_EXR_COMPRESSION);
+  params.push_back(IMWRITE_EXR_COMPRESSION_PIZ);
+
+#ifndef GENERATE_DATA
+  const Mat img = cv::imread(filenameInput, IMREAD_UNCHANGED);
+#else
+  const Size sz(64, 32);
+  Mat img(sz, CV_32FC1, Scalar(0.5, 0.1, 1));
+  img(Rect(10, 5, sz.width - 30, sz.height - 20)).setTo(Scalar(1, 0, 0));
+  ASSERT_TRUE(cv::imwrite(filenameInput, img, params));
+#endif
```

Comment:
The else branch is not needed any more as soon as test reads test image and checks output file size. Else branch should not pass test.

Reply:
"Else" branch is for initial test data generation.

This code is useful for test data update in the future.